ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+
+`SET_CODE_TX_TYPE=0x04` gives EOA the ability to execute code. Executing from the EOA brings some benefits:
+
+* Aligned with the nature of EOA that the execution is fully controlled by the account owner. EOA owner can easily turn
+  off all smart contract features by un-delegating the contract code.
+* `msg.sender` is always EOA address
+* Execution code can be stateless, this allows `account` to store no data.
+
+In the case that the EOA doesn't need to execute its contract code, or the intent cost is too high. The owner can use
+the account as EOA.
+
+### Validation in standard contract
+
+Validation logic commonly relies on contract state, for example, weighted multi-owner signature needs to track the
+weight of each signer. Keeping the functionality of intent validate purely inside `IStandard` makes it similar to
+the `EntryPoint` concept in ERC-4337, but a simpler version. Standard only taking responsibility of validation makes it
+easier for contract engineer to build, audit and maintain.
+
+Also, the `IStandard` interface can be considered as "modular". There can be a "compound" standard that
+breaks one intent into smaller pieces and call other standards to validate each piece and later combine the results
+together.
+
+### Gas abstraction
+
+This design allows any address to initiate transaction on behalf of the intent's sender, providing a gasless experience. The sender can define the "payment" to the sender in intent header or instruction, with transfers any token from the sender's account to `tx.origin`.
+
+### Auditability of both validation and execution
+
+It is very important both the standard and account implementation can be publicly audited and shared. The most important
+reason is security. And this can also help mediate the compatibility issues between standard and account.
+
+### Solver is relayer, relayer is paymaster, paymaster is bundler
+
+Within a intent-centric system, solvers are helping account owners to fulfill the intent and solvers are rewarded. This
+proposal allows any solver to execute the intent, bringing a positive competitive environment. And supported by gas
+abstraction, solvers will pay the native token as gas fee and take other tokens back from the EOA account. Besides,
+solvers can further reduce the cost of their side by bundling multiple intent executions into one blockchain
+transaction.
+
+Each solver needs to develop its own strategy to maximize its profit. And this EIP doesn't define anything about how a
+solver executes the intents, meaning there is no limitation.
+
+### Intents can be rewarded, repeatable
+
+And another fact, which is easy to be ignored, is that intent has value by itself. For example, if an EOA is always
+willing to swap 1000 USDC to 1000 USDT and vice versa, this EOA will be considered as a "liquidity provider" in the
+market. The account can send the signed intent to exchanges and make the exchange reward the account every time the
+intent is executed.
```

Comment:
This is again more relevant to Motivation. You aren't explaining any choices made within this document, so it doesn't belong in Rationale.

Reply:
Updated to be `No re-entry protection enforced`